### ISSUE TITLE
Unhandled event: CHANNEL_PINS_UPDATE

### ISSUE THREAD

**Issue by skull132** [Rapptz/discord.py#300]
Was looking at the logs for my bot, and kept seeing unhandled event messages. I scrounged up some debug data about these events, and have determined the following:
1. A `MESSAGE_UPDATE` event is sent **before** a `CHANNEL_PINS_UPDATE` event is sent. This includes the standard message data, etcetera.
   - At this point, the data reflects the new `pinned` status of the message that was "edited" and pinned/unpinned, so if you pin a message, it's set to `True`, and if you unpin it, it's set to `False`.
2. The `CHANNEL_PINS_UPDATE` event is sent right afterwards, and includes the following data:
   - `channel_id` - the ID of the channel which had its pinned messages updated.
   - `last_pin_timestamp` - the timestamp of when the last message in the **updated** list of pinned messages was pinned. Meaning, if a pin is added, it'll showcase the timestamp for that pin (as it's the last one on the list). If a channel has only one pin, and that pin is removed, then this value will be `None`.
   - The `last_pin_timestamp` appears to be the timestamp of when the last pinned message was pinned, and not the timestamp of when the pinned message was sent.

I'm tempted to look into adding an event handler for this myself, but since that may fail (and I won't be able to try for a week), I figured I'll throw this up here just in case.

Raw data to follow.
**Pin added to no pins:**

```
2016-08-14 18:03:03,919:DEBUG:discord.client: Dispatching event socket_raw_receive
2016-08-14 18:03:03,919:DEBUG:discord.gateway: WebSocket Event: {'s': 11, 'd': {'pinned': True, 'mention_roles': [], 'mentions': [], 'author': {'discriminator': '1984', 'id': '84559773487353856', 'username': 'Skull132', 'avatar': '0d5b8e5ac1e4e41b07ab4fd524c7205e'}, 'timestamp': '2016-08-14T15:01:18.782000+00:00', 'channel_id': '140235945508208640', 'tts': False, 'attachments': [], 'embeds': [], 'mention_everyone': False, 'id': '214398051421388800', 'content': 'test', 'type': 0, 'edited_timestamp': None}, 't': 'MESSAGE_UPDATE', 'op': 0}
2016-08-14 18:03:03,919:DEBUG:discord.client: Dispatching event socket_response
2016-08-14 18:03:03,919:DEBUG:discord.client: Dispatching event message_edit
2016-08-14 18:03:03,919:DEBUG:discord.client: Dispatching event socket_raw_receive
2016-08-14 18:03:03,919:DEBUG:discord.gateway: WebSocket Event: {'s': 12, 'd': {'channel_id': '140235945508208640', 'last_pin_timestamp': '2016-08-14T15:03:09.029034+00:00'}, 't': 'CHANNEL_PINS_UPDATE', 'op': 0}
2016-08-14 18:03:03,919:DEBUG:discord.client: Dispatching event socket_response
2016-08-14 18:03:03,919:INFO:discord.gateway: Unhandled event CHANNEL_PINS_UPDATE
```

**Pin removed leaving no pins:**

```
2016-08-14 18:04:05,885:DEBUG:discord.client: Dispatching event socket_raw_receive
2016-08-14 18:04:05,885:DEBUG:discord.gateway: WebSocket Event: {'s': 15, 'd': {'pinned': False, 'mention_roles': [], 'mentions': [], 'author': {'discriminator': '1984', 'id': '84559773487353856', 'username': 'Skull132', 'avatar': '0d5b8e5ac1e4e41b07ab4fd524c7205e'}, 'timestamp': '2016-08-14T15:01:18.782000+00:00', 'channel_id': '140235945508208640', 'tts': False, 'attachments': [], 'embeds': [], 'mention_everyone': False, 'id': '214398051421388800', 'content': 'test', 'type': 0, 'edited_timestamp': None}, 't': 'MESSAGE_UPDATE', 'op': 0}
2016-08-14 18:04:05,885:DEBUG:discord.client: Dispatching event socket_response
2016-08-14 18:04:05,885:DEBUG:discord.client: Dispatching event message_edit
2016-08-14 18:04:05,885:DEBUG:discord.client: Dispatching event socket_raw_receive
2016-08-14 18:04:05,885:DEBUG:discord.gateway: WebSocket Event: {'s': 16, 'd': {'channel_id': '140235945508208640', 'last_pin_timestamp': None}, 't': 'CHANNEL_PINS_UPDATE', 'op': 0}
2016-08-14 18:04:05,886:DEBUG:discord.client: Dispatching event socket_response
2016-08-14 18:04:05,886:INFO:discord.gateway: Unhandled event CHANNEL_PINS_UPDATE
```

**Comment by Rapptz** [comment]
I knew about this event and intentionally didn't handle it. The event is honestly a tad bit useless for use. There's no good way to make an API out of it and the only reason it exists is to make the red dot appear in the pin button in the client. e.g. 

![](http://i.imgur.com/skcvgsH.png)

**Comment by Avaren** [comment]
Implemented in #451
